src/models/concrete_syntax.rs

```diff
@@ -153,35 +219,49 @@ pub(crate) fn get_matches_for_node(
       let mut tmp_cursor = cursor.clone();
       let current_node = cursor.node();
       let current_node_code = current_node.utf8_text(source_code).unwrap();
-      find_next_sibling(&mut tmp_cursor);
-
-      // Support for trailing commas
-      // This skips trailing commas as we are parsing through the match template
-      // Skips the comma node if the template doesn't contain it.
-      let next_node = tmp_cursor.node();
-      let next_node_text = next_node.utf8_text(source_code).unwrap();
-      if next_node_text == "," && !meta_advanced.0.starts_with(',') {
-        find_next_sibling(&mut tmp_cursor); // Skip comma
-      }
 
-      if let (mut recursive_matches, true) =
-        get_matches_for_node(&mut tmp_cursor, source_code, &meta_advanced)
-      {
-        // If we already matched this variable, we need to make sure that the match is the same. Otherwise, we were unsuccessful.
-        // No other way of unrolling exists.
-        if recursive_matches.contains_key(var_name)
-          && recursive_matches[var_name].text.trim() != current_node_code.trim()
-        {
-          return (HashMap::new(), false);
+      let mut should_match = find_next_sibling(&mut tmp_cursor);
```

Comment:
determine whether the next recursive call to `get_matches_for_node` should match any node. If there are no "siblings" left, then we should not match anything